apps/web/src/components/Feed/FeedSuggestedProfileSection.tsx

```diff
@@ -0,0 +1,206 @@
+  const [profiles, setProfiles] = useState([]);
+
+  useEffect(() => {
+    const users = [];
+
+    for (const edge of query.viewer.suggestedUsers?.edges ?? []) {
+      if (edge?.node) {
+        users.push(edge.node);
+      }
+    }
+
+    setProfiles(users);
+  }, [query.viewer.suggestedUsers?.edges]);
+
+  const slideContent = useMemo(() => {
+    const rows = [];
+
+    for (let i = 0; i < profiles.length; i += itemsPerSlide) {
+      const row = profiles.slice(i, i + itemsPerSlide);
+      const rowElements = (
+        <StyledSuggestedProfileSet gap={isMobileOrMobileLargeWindow ? 4 : 16}>
+          {row.map((profile, idx) => (
+            <SuggestedProfileCard
+              key={idx}
+              queryRef={query}
+              userRef={profile}
+              variant="compact"
+              onClick={() => handleProfileClick(profile?.username)}
+            />
+          ))}
+          {row.length < itemsPerSlide && <StyledPlaceholder />}
+        </StyledSuggestedProfileSet>
+      );
+
+      rows.push(rowElements);
+    }
+
+    return rows;
+  }, [itemsPerSlide, profiles, isMobileOrMobileLargeWindow, handleProfileClick, query]);
+
+  if (!query.viewer?.suggestedUsers) {
+    return null;
+  }
+
+  return (
+    <FeedSuggestedProfileSectionContainer gap={isMobileOrMobileLargeWindow ? 12 : 16}>
+      <StyledTitleContainer>
+        <StyledTitle>Suggested creators and collectors</StyledTitle>
+      </StyledTitleContainer>
+      <StyledContainer gap={16}>
+        <Carousel
+          slideContent={slideContent}
+          loop={isMobileOrMobileLargeWindow ? false : true}
+          variant={isMobileOrMobileLargeWindow ? 'compact' : 'default'}
+        />
+      </StyledContainer>
+    </FeedSuggestedProfileSectionContainer>
+  );
+}
+
+const StyledContainer = styled(VStack)`
+  margin-left: -56px;
+  @media only screen and ${breakpoints.tablet} {
+    flex-direction: row;
+    gap: 16px;
+    margin-left: 0px;
+    width: 100%;
+    overflow-x: hidden;
+  }
+`;
+
+const StyledTitleContainer = styled(HStack)`
+  padding-top: 42px;
+
+  @media only screen and ${breakpoints.desktop} {
+    padding-top: 28px;
+    padding-left: 46px;
+  }
+`;
+
+const StyledTitle = styled(BaseM)`
+  font-size: 14px;
+  font-weight: 700;
+  line-height: 20px;
+`;
+
+const StyledSuggestedProfileSet = styled(HStack)`
+  width: 100%;
+`;
+
+const StyledPlaceholder = styled.div`
+  width: 100%;
+`;
+
+type FeedSuggestedProfileSectionWithBoundaryProps = {
+  queryRef: FeedSuggestedProfileSectionWithBoundaryFragment$key;
+};
+
+export default function FeedSuggestedProfileSectionWithBoundary({
+  queryRef,
+}: FeedSuggestedProfileSectionWithBoundaryProps) {
+  const query = useFragment(
+    graphql`
+      fragment FeedSuggestedProfileSectionWithBoundaryFragment on Query {
+        ...FeedSuggestedProfileSectionQueryFragment
+      }
+    `,
+    queryRef
+  );
+  return (
+    <ReportingErrorBoundary fallback={<></>}>
```

Comment:
should we have skeleton states?

Reply:
right I was just following the structure for the other feed items and they didn't have a `ReportingErrorBoundary` or `Suspense` fallback (attached sc of one). So do you think we should add the skeleton state here for this section?

<img width="752" alt="Screenshot 2024-04-17 at 14 49 22" src="https://github.com/gallery-so/gallery/assets/49758803/6202d902-e78b-47b1-ba38-4c952e56f024">